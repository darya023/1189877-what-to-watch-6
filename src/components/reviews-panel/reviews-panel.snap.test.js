import React from 'react';
import {render} from '@testing-library/react';
import ReviewsPanel from './reviews-panel';

const fakeReviews = [
  {
    text: `Test 1`,
    date: `2021-02-23T08:04:28.658Z`,
    id: `1`,
    rating: 3.2,
    authorId: `19`,
    authorName: `Test1`
  },
  {
    text: `Test 2`,
    date: `2020-02-03T08:00:28.000Z`,
    id: `1`,
    rating: 7,
    authorId: `2`,
    authorName: `Test2`
  }
];

describe(`Test for ReviewsPanel`, () => {
  it(`Should ReviewsPanel render correctly`, () => {
    const {container} = render(
        <ReviewsPanel
          reviews={fakeReviews}
        />
    );
    expect(container).toMatchSnapshot();
  });
  it(`Should ReviewsPanel render correctly without reviews`, () => {
    const {container} = render(
        <ReviewsPanel
          reviews={[]}
        />
    );
    expect(container).toMatchSnapshot();
  });
});
