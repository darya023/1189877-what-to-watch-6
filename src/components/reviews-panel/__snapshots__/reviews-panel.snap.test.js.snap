// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ReviewsPanel Should ReviewsPanel render correctly 1`] = `
<div>
  <div
    class="movie-card__reviews movie-card__row"
  >
    <div
      class="movie-card__reviews-col"
    >
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            Test 1
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              Test1
            </cite>
            <time
              class="review__date"
              datetime="2021-02-23"
            >
              February 23, 2021
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          3.2
        </div>
      </div>
    </div>
    <div
      class="movie-card__reviews-col"
    >
      <div
        class="review"
      >
        <blockquote
          class="review__quote"
        >
          <p
            class="review__text"
          >
            Test 2
          </p>
          <footer
            class="review__details"
          >
            <cite
              class="review__author"
            >
              Test2
            </cite>
            <time
              class="review__date"
              datetime="2020-02-03"
            >
              February 3, 2020
            </time>
          </footer>
        </blockquote>
        <div
          class="review__rating"
        >
          7
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test for ReviewsPanel Should ReviewsPanel render correctly without reviews 1`] = `
<div>
  <div
    class="movie-card__reviews movie-card__row"
  >
    <div
      class="movie-card__reviews-col"
    >
      <div
        class="review"
      >
        There are no reviews yet
      </div>
    </div>
  </div>
</div>
`;
